README.md

```diff
@@ -32,75 +32,53 @@ return [
 
-#### Define your features
-
-First you have to configure the bundle and define your feature flags in the `config.yml`.
+To configure and register a feature manager you need a factory service. You may also need to change some options to the factory.
 
 ```yaml
 # ...
 novaway_feature_flag:
     default_manager: default
     managers:
         default:
-            storage: 'Novaway\Bundle\FeatureFlagBundle\Storage\ArrayStorage'
+            factory: 'novaway_feature_flag.factory.array'
             options:
                 features:
                     my_feature_1: false
                     my_feature_2: true
                     my_feature3: '%env(bool:FEATURE_ENVVAR)%'
 ```
 
-The `Novaway\Bundle\FeatureFlagBundle\Storage\ArrayStorage` allows you to define your feature flags in an extended way:
+The factories that come with this bundle can be found in the table below.
+
+| Factory service id                 | Options    |
+|------------------------------------|------------|
+| novaway_feature_flag.factory.array | `features` |
+
+#### Example configuration
 
 ```yaml
 # ...
 novaway_feature_flag:
     default_manager: default
     managers:
         default:
-            storage: 'Novaway\Bundle\FeatureFlagBundle\Storage\ArrayStorage'
+            factory: novaway_feature_flag.factory.array
             options:
                 features:
-                    my_feature_1:
-                        enabled: false
-                        description: MyFeature1 description text
-                    my_feature_2:
-                        enabled: true
-                        description: MyFeature2 description text
-                    my_feature3:
-                        enabled: '%env(bool:FEATURE_ENVVAR)%'
-                        description: MyFeature3 description text
-```
-
-You can declare multiple managers. Multiple providers is useful if you want to use different storage providers or to isolate your features flags.
-
-```yaml
-# ...
-novaway_feature_flag:
-    default_manager: manager_foo
-    managers:
-        manager_foo:
-            storage: 'Novaway\Bundle\FeatureFlagBundle\Storage\ArrayStorage'
+                    my_feature_1: []
+                    my_feature_2: ~
+                    my_feature_3: false
+        default2:
+            factory: novaway_feature_flag.factory.array
             options:
                 features:
-                    my_feature_1:
-                        enabled: false
-                        description: MyFeature1 description text
-                    my_feature_2:
+                    my_feature_4:
                         enabled: true
-                        description: MyFeature2 description text
-                    my_feature3:
+                        description: 'Lorem Ipsum'
+                    my_feature_5:
                         enabled: '%env(bool:FEATURE_ENVVAR)%'
-                        description: MyFeature3 description text
-        manager_bar:
```

Comment:
The documentation refers to this manager.

You should update existing references.